doc/architecture.txt

```diff
@@ -0,0 +1,32 @@
+/**
+@page architecture SDK Architecture and Design
+@brief This page documents the architecture and design goals of this SDK.
+
+The Embedded C SDK has a layered architecture. It is currently based on the MQTT protocol, but may support protocols like HTTP in the future.
+
+This SDK is primarily targeted for resource-constrained devices.
+
+@section architecture_considerations Considerations during design
+- Low memory footprint (around 6k of code size without TLS library for the MQTT layer)
+- Flexibility in picking and choosing functionality
+- Static memory only (no malloc)
+- Configurable resource usage (JSON tokens, MQTT subscription handlers, etc.)
+- Portability across RTOS
+
+@section architecture_diagram Layered architecture
+@brief There are primarily 3 layers in the SDK as shown in the diagram below.
+
+@image html architecture.png
+
+@subsection architecture_tls TLS layer
+
+This SDK uses TLS sockets to talk to the server. We provide wrappers around mbed TLS (ARM) that allow the libraries to connect with AWS IoT Platform.
+
+@subsection architecture_mqtt AWS IoT MQTT client
+
+The MQTT library included in this SDK is a modification of the Paho Embedded C client. MQTT provides a publish subscribe protocol for communication with AWS IoT. It can be used standalone or with Thing Shadows.
```

Comment:
1. Should this officially mention about being a modification of Paho? (If yes, a link to the Paho MQTT library would be valuable).
2. The _MQTT provides a publish subscribe protocol for communication with AWS IoT_ sentence can be misleading to suggest that MQTT protocoal is _only_ for AWS IoT. 
Maybe, the sentence can be updated to mention something like "MQTT is a publish-subscribe communication protocol supported by AWS IoT"
3. "_It can be used standalone or with Thing Shadows._" What about Jobs? This sentence could be misinterpreted about not being available to AWS IoT services other than Shadow.

Reply:
Fixed